src/backend/distributed/commands/utility_hook.c

```diff
@@ -924,6 +956,49 @@ ShouldAddNewTableToMetadata(Node *parsetree)
 }
 
 
+/*
+ * ServerUsesPostgresFDW gets a foreign server name and returns true if the FDW that
+ * the server depends on is postgres_fdw. Returns false otherwise.
+ */
+static bool
+ServerUsesPostgresFDW(char *serverName)
+{
+	ForeignServer *server = GetForeignServerByName(serverName, false);
+	ForeignDataWrapper *fdw = GetForeignDataWrapper(server->fdwid);
+
+	if (strcmp(fdw->fdwname, "postgres_fdw") == 0)
+	{
+		return true;
+	}
+
+	return false;
+}
+
+
+/*
+ * ErrorIfOptionListHasNoTableName gets an option list (DefElem) and errors out
+ * if the list does not contain a table_name element.
+ */
+static void
+ErrorIfOptionListHasNoTableName(List *optionList)
+{
+	char *table_nameString = "table_name";
+	DefElem *option = NULL;
+	foreach_ptr(option, optionList)
+	{
+		char *optionName = option->defname;
+		if (strcmp(optionName, table_nameString) == 0)
+		{
+			return;
+		}
+	}
+
```

Comment:
needs test

Reply:
Added: https://github.com/citusdata/citus/pull/5461/files#diff-9510e755caf2a2e6f76ae30b2379705372a53c66325c5c655297a93b766d0eb1R416

Probably codecov comments are not up-to-date